### DISCUSSION TITLE
Consider including instaparse

### DISCUSSION THREAD

**Discussion by borkdude** [babashka/babashka#1335] [Ideas]
Upvote or comment if you're interested. It would also be interesting into looking if instaparse could be executed from source.

**Comment by frou** [comment]
This would be a great addition, because parsing weird and wonderful textual output of other commands is pretty much the core tenet of Unix scripting.

**Comment by 7tupel** [comment]
That would be great. It would allow building a parser with Babashka and instaparse to validate commit messages to follow the conventional commit specs. I've had this in my mind for a while now as I'm trying to build a full development and CI/CD workflow in Clojure and Babashka instead of mixing it with other tools i.e. based on Node.js.

**Reply by 7tupel** [reply]
Just for the curious: I build the parser for Commit Messages and some other tooling in my little [clci](https://github.com/ClockworksIO/clci/) library.

This is still a work in progress and I expect several bugs are still hiding in there!

**Comment by colin-p-hill** [comment]
Currently running into a use case where I'm trying to parse the output of low-level git commands. I can do this by hand, but instaparse would make it a heck of a lot easier.

**Reply by borkdude** [reply]
Meanwhile you could maybe use https://github.com/rm-hull/jasentaa - it works in bb

**Reply by colin-p-hill** [reply]
Well dang, I was working off the gist that's based on. My searches didn't turn up any full implementations, for some reason. Thanks!

**Comment by borkdude** [comment]
I've made a pod + wrapping library here:

https://github.com/babashka/instaparse.bb

This can be considered an initial step in playing around with Instaparse + bb. If the pod ends up being spectacularly popular, I'd have a better idea if Instaparse should be included into bb itself at one point.

**Comment by jave** [comment]
I tried making a jaasentaa parser 1st, failed, and then tried instaparse, that worked somewhat better.

In my case I wanted to make a parser for "logkeys" files, which I use in order to improve my qmk keyboard layout,
so its not mission critical.

**Reply by borkdude** [reply]
Have you tried https://github.com/babashka/instaparse.bb?

**Reply by jave** [reply]
no, just clojure. I can try the pod and see if it works for me
